From 074178fed01bbac4cc3a5e3f90c573614e4de179 Mon Sep 17 00:00:00 2001
From: Antoine Jacoutot <ajacoutot@openbsd.org>
Date: Tue, 24 May 2011 11:15:08 +0000
Subject: Don't try to close a non-opened fd

On OpenBSD we do not use /proc. As a result, gnome-power-manager segfaults on
startup due to a reverse logic in the code.

Signed-off-by: Richard Hughes <richard@hughsie.com>
---
diff --git a/src/gpm-load.c b/src/gpm-load.c
index eab0129..3b0f572 100644
--- a/src/gpm-load.c
+++ b/src/gpm-load.c
@@ -192,7 +192,7 @@ gpm_load_get_cpu_values (long unsigned *cpu_idle, long unsigned *cpu_total)
 	*cpu_total = cpu_user + cpu_nice + cpu_system + *cpu_idle;
 	ret = TRUE;
 out:
-	if (!fd)
+	if (fd)
 		fclose (fd);
 	return ret;
 }
--
cgit v0.9
